src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -110,12 +110,33 @@ internal partial class Binder
 
             if (conversion.Kind == ConversionKind.SwitchExpression)
             {
-                return ConvertSwitchExpression((BoundUnconvertedSwitchExpression)source, destination, conversionIfTargetTyped: conversion, diagnostics);
+
+                var convertedSwitch = ConvertSwitchExpression((BoundUnconvertedSwitchExpression)source, destination, conversionIfTargetTyped: conversion, diagnostics);
+                return new BoundConversion(
```

Comment:
Can we move this `new BoundConversion` into `ConvertSwitchExpression`? (would be conditional on having a `conversionIfTargetTyped`)
I think that would help reduce sprawl in the current method.

Same for `ConvertConditionalExpression` below.

Reply:
I intentionally didn't do that, because `ConvertSwitchExpression` is used in other places.

---
In reply to: [595617069](https://github.com/dotnet/roslyn/pull/51791#discussion_r595617069) [](ancestors = 595617069)